Azure-Samples/aks-store-demo
# Migrate to GWAPI for store-front routing

## Purpose

* Migrate from Istio-specific Gateway and VirtualService resources to the standardized Gateway API resources (Gateway and HTTPRoute). This aligns with the new stable [Gateway API version](https://istio.io/latest/blog/2024/gateway-mesh-ga/) and ensures consistency across environments.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Test the code

```
1. Deploy the updated Gateway and HTTPRoute configurations to the Kubernetes cluster.
2. Ensure the store-front service is running in the 'pets' namespace.
3. Verify that the Gateway is correctly listening on port 80 and handling HTTP requests.
4. Test HTTP requests to confirm they are routed to the store-front service as expected.
```

## What to Check
Verify that the following are valid
* The Gateway is created successfully and is listening on port 80.
* HTTP requests to the Gateway are correctly routed to the store-front service.
* The HTTPRoute is correctly attached to the Gateway and matches all hostnames and paths.
* No errors or unexpected behavior occurs during routing.

## Other Information

* Verify that the cluster has the appropriate permissions to manage Gateway and HTTPRoute resources.
